mos_tests/neutron/python_tests/test_l3_ha.py

```diff
@@ -474,3 +504,162 @@ def test_ban_active_l3_agent_with_external_connectivity(self, router,
+            device_id=router_id)['ports']
+        active_l3_ha_port_for_router_id = [
+            port for port in ports_list_for_router
+            if port['binding:host_id'] == active_hostname][0]['id']
+        active_ha_iface_id = 'ha-{}'.format(
+            active_l3_ha_port_for_router_id[:11])
+
+        # Move down router ha-port
+        with self.background_ping(
+                vm=instance,
+                vm_keypair=self.instance_keypair,
+                ip_to_ping=settings.PUBLIC_TEST_IP) as ping_result:
+            with active_node.ssh() as remote:
+                logger.info("Move down ha-port on router")
+                remote.check_call(
+                    "ip netns exec qrouter-{router_id} "
+                    "ip link set dev {iface_id} down".format(
+                        router_id=router_id, iface_id=active_ha_iface_id))
+                self.wait_router_rescheduled(
+                    router_id=router['router']['id'],
+                    from_node=active_hostname)
+
+        assert (ping_result['sended'] - ping_result['received']) < 10
+
+    def test_ban_l3_agent_with_tcpdump_check(self, router, prepare_openstack):
+        """Ban l3 active agent and check it by tcpdump log.
+
+         Steps:
+            1. Create network net01, subnet net01_subnet
+            2. Create router with gateway to external net and
+               interface with net01
+            3. Launch instance and associate floating IP
+            4. Start tcpdump on all controllers
+            5. Check ping from external host to instance by floating IP
+            6. Ban active l3 agent
+            7. Wait until router rescheduled
+            8. Stop ping
+            9. Stop tcpdump
+            10. Check that tcpdump results and active l3 agents statuses
+            11. Check that ping lost less than 10 packets
+        """
+        def start_tcpdump_in_background(node, router_id, active_qg_iface_id):
+            """Start tcpdump on l3ha-router provided port.
+
+            :param node: node on which start tcpdump
+            :param router_id: router id on which start tcpdump
+            :param active_qg_iface_id: interface id on which start tcpdump
+            :return: thread with executing command over ssh
+            """
+            logger.info('Start tcpdump on {0}'.format(node.data['fqdn']))
+            cmd_for_tcpdump = (
+                'ip netns exec qrouter-{router_id} '
+                'tcpdump -n -l -i {qg_port_id} icmp 2>&1 | '
+                'tee /tmp/tcpdump.log'.format(
+                    router_id=router_id, qg_port_id=active_qg_iface_id))
+
+            def start_tcpdump():
+                with node.ssh() as remote:
+                    res = remote.execute(cmd_for_tcpdump)
+                    return res
+
+            tcpdump_thread = threading.Thread(target=start_tcpdump)
+            tcpdump_thread.start()
+            return tcpdump_thread
+
+        def analyze_tcpdump(node):
```

Comment:
may be better call this function like get_last_icmp_packet (or what you returning here)